libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/ui/RouteLineComponent.kt

```diff
@@ -10,61 +10,37 @@ import com.mapbox.maps.plugin.gestures.gestures
 import com.mapbox.maps.plugin.locationcomponent.OnIndicatorPositionChangedListener
 import com.mapbox.maps.plugin.locationcomponent.location
 import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
-import com.mapbox.navigation.base.route.NavigationRoute
 import com.mapbox.navigation.core.MapboxNavigation
 import com.mapbox.navigation.core.internal.extensions.flowRouteProgress
-import com.mapbox.navigation.core.internal.extensions.flowRoutesUpdated
+import com.mapbox.navigation.core.lifecycle.MapboxNavigationObserver
 import com.mapbox.navigation.ui.base.lifecycle.UIComponent
+import com.mapbox.navigation.ui.maps.internal.ActiveRouteLineComponentContract
+import com.mapbox.navigation.ui.maps.internal.RouteLineComponentContract
 import com.mapbox.navigation.ui.maps.route.line.MapboxRouteLineApiExtensions.findClosestRoute
 import com.mapbox.navigation.ui.maps.route.line.MapboxRouteLineApiExtensions.setNavigationRouteLines
 import com.mapbox.navigation.ui.maps.route.line.api.MapboxRouteLineApi
 import com.mapbox.navigation.ui.maps.route.line.api.MapboxRouteLineView
 import com.mapbox.navigation.ui.maps.route.line.model.MapboxRouteLineOptions
 import com.mapbox.navigation.ui.maps.route.line.model.NavigationRouteLine
-import com.mapbox.navigation.ui.utils.internal.Provider
 import com.mapbox.navigation.utils.internal.ifNonNull
-import kotlinx.coroutines.flow.SharingStarted
 import kotlinx.coroutines.flow.collect
-import kotlinx.coroutines.flow.map
-import kotlinx.coroutines.flow.stateIn
 import kotlinx.coroutines.launch
 
-@ExperimentalPreviewMapboxNavigationAPI
-interface RouteLineComponentContract {
-    fun setRoutes(mapboxNavigation: MapboxNavigation, routes: List<NavigationRoute>)
-}
-
-@ExperimentalPreviewMapboxNavigationAPI
-internal class MapboxRouteLineComponentContract : RouteLineComponentContract {
-    override fun setRoutes(mapboxNavigation: MapboxNavigation, routes: List<NavigationRoute>) {
-        mapboxNavigation.setNavigationRoutes(routes)
-    }
-}
-
 @ExperimentalPreviewMapboxNavigationAPI
 class RouteLineComponent(
     private val mapboxMap: MapboxMap,
     private val mapPlugins: MapPluginProviderDelegate,
     private val options: MapboxRouteLineOptions,
     private val routeLineApi: MapboxRouteLineApi = MapboxRouteLineApi(options),
     private val routeLineView: MapboxRouteLineView = MapboxRouteLineView(options),
-    contractProvider: Provider<RouteLineComponentContract>? = null
+    contract: RouteLineComponentContract? = null
```

Comment:
By injecting a `Provider` we defer the creation of a `Contract` until the component is attached. Why the change?

Reply:
The contract does not work as it is, so I just made it work and this was not necessary.

Why is the `Provider<` required and a why have lazy initialization required here? We could add `Provider< MapboxRouteLineOptions` as well - but why would we?

Reply:
It actually, makes more sense for the options to be a provider than the contract. The contract, typically does not have any data associated with it. So why would it have a lazy initialization? The contract itself can define how it provides a definition 🤔